Packs/Code42/Integrations/Code42/Code42.py

```diff
@@ -15,425 +18,499 @@
-        alerts = self._sdk.security.alerts
         try:
-            res = alerts.get_query_details([alert_id])
+            res = self._sdk.alerts.get_details(alert_id)
         except Exception:
             return None
-        else:
-            # There will only ever be one alert since we search on just one ID
-            return res.json().get('alerts')[0]
+        return res["alerts"][0]
 
     def get_current_user(self):
         try:
             res = self._sdk.users.get_current_user()
         except Exception:
             return None
-        return res.json()
+        return res
 
     def remove_user_from_departing_employee(self, username):
         try:
-            de = self._sdk.employee_case_management.departing_employee
-            res = de.get_case_by_username(username)
+            user_id = self.get_user_id(username)
+            self._sdk.detectionlists.departing_employee.remove(user_id)
         except Exception:
             return None
-        case_id = res.json().get('caseId')
+        return user_id
+
+    def resolve_alert(self, id):
         try:
-            de.resolve_departing_employee(case_id)
+            self._sdk.alerts.resolve(id)
         except Exception:
             return None
-        return case_id
+        return id
 
-    def resolve_alert(self, id):
-        alerts = self._sdk.security.alerts
+    def get_user_id(self, username):
         try:
-            alerts.resolve_alert(id)
+            res = self._sdk.users.get_by_username(username)
         except Exception:
             return None
-        return id
+        return res["users"][0]["userUid"]
 
     def search_json(self, payload):
         try:
-            res = self._sdk.security.search_file_events(payload)
+            res = self._sdk.securitydata.search_file_events(payload)
         except Exception:
             return None
-        return res.json().get('fileEvents')
+        return res["fileEvents"]
 
 
 @logger
 def build_query_payload(args):
     """
     Build a query payload combining passed args
     """
+    _hash = args.get("hash")
+    hostname = args.get("hostname")
+    username = args.get("username")
+    exposure = args.get("exposure")
+
     search_args = []
-    if args.get('hash'):
-        if len(args['hash']) == 32:
-            search_args.append(MD5.eq(args['hash']))
-        elif len(args['hash']) == 64:
-            search_args.append(SHA256.eq(args['hash']))
-    if args.get('hostname'):
-        search_args.append(OSHostname.eq(args['hostname']))
-    if args.get('username'):
-        search_args.append(DeviceUsername.eq(args['username']))
-    if args.get('exposure'):
-        # Because the CLI can't accept lists, convert the args to a list if the type is string.
-        if isinstance(args['exposure'], str):
-            args['exposure'] = args['exposure'].split(',')
-        search_args.append(ExposureType.is_in(args['exposure']))
+    hash_filter = _create_hash_filter(_hash)
+    if hash_filter:
```

Comment:
Note there linter prevents us from doing `not hash_filter or search_args.append(hash_filter)`

Reply:
Just a thought, to make it look good !! we can have an inner function as follows, with a better function name:

```
def append_search_args(field, action):
      if field:
           search_args.append(action)

append_search_args(hash_filter, hash_filter)
append_search_args(hostname, OSHostname.eq(hostname))
append_search_args(username, DeviceUsername.eq(username))
append_search_args(exposure, _create_exposure_filter(exposure))
```